### PR TITLE
Eli/appeals 27096

### PR DESCRIPTION
**Pull request by elilogbro**
Resolves [APPEALS-27096](https://jira.devops.va.gov/browse/APPEALS-27096)

# Description
As a developer, I need to:
1. Update the make command 'reseed-vbms-ext-claim' to include 'external-db-create', creating the vbms_ext_claim table if it isn't standing
2. Change the name of BatchProcessPriorityEpSync to PriorityEpSyncBatchProcess
3. Change the name of BatchProcessPriorityEpSyncJob to PriorityEpSyncBatchProcssJob
4. Implement a redis mutex distributed lock around the creation of batches within the PriorityEpSyncBatchProcessJob
5. Add 1-hr looping functionality which sleeps 5s between iterations to PriorityEpSyncBatchProcessJob
6. Add 1-hr looping functionality which sleeps 5s between iterations to PopulateEndProductSyncQueueJob
7. Create a migration that adds "remediated" boolean column that defaults to false, "remediation_notes" text column, and "updated_at" datetime column to caseflow_stuck_records
8. Update vbms_ext_claim factory to default claim_id to "300000" if there aren't any records on the table

## Acceptance Criteria
- [x] Update make command 'reseed-vbms-ext-claim' to ensure it doesn't error out if vbms_ext_claim isn't standing
- [x] Update naming of BatchProcessPriorityEpSync to PriorityEpSyncBatchProcess
- [x] Update naming of BatchProcessPriorityEpSyncJob to PriorityEpSyncBatchProcessJob
- [x] Add a distributed lock on the creation of batches in PriorityEpSyncBatchProcessJob
- [x] Add 1-hr loop with 5s sleep to PriorityEpSyncBatchProcessJob
- [x] Add 1-hr loop with 5s sleep to PopulateEndProductSyncQueueJob
- [x] Create a migration that adds "remediated" boolean column that defaults to false, "remediation_notes" text column, and "updated_at" datetime column to caseflow_stuck_records
- [x] Update vbms_ext_claim factory to default claim_id to "300000" if there aren't any records on the table

## Testing Plan
1. Testing Steps
https://jira.devops.va.gov/browse/APPEALS-24888
https://jira.devops.va.gov/browse/APPEALS-27410
5. Test Execution
https://jira.devops.va.gov/browse/APPEALS-27461
https://jira.devops.va.gov/browse/APPEALS-27487
9. XRAY
https://jira.devops.va.gov/secure/XrayExecuteTest!default.jspa?testExecIssueKey=APPEALS-27461&testIssueKey=APPEALS-24888
https://jira.devops.va.gov/secure/XrayExecuteTest!default.jspa?testExecIssueKey=APPEALS-27487&testIssueKey=APPEALS-27410

# Backend
## Database Changes
*Only for Schema Changes*
* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [x] Verify that `migrate:rollback` works as desired

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec

### CODE REVIEW

**Review by elilogbro — COMMENTED** [review]
After looking at this again, I believe we can revert first_batch_process and second_batch_process variables to BatchProcess.first and BatchProcess.second

File: spec/jobs/batch_processes/priority_ep_sync_batch_process_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,207 @@
+# frozen_string_literal: true
+
+require "./app/jobs/batch_processes/priority_ep_sync_batch_process_job.rb"
+require "./app/models/batch_processes/batch_process.rb"
+
+describe PriorityEpSyncBatchProcessJob, type: :job do
+  let!(:syncable_end_product_establishments) do
+    create_list(:end_product_establishment, 99, :active_hlr_with_cleared_vbms_ext_claim)
```
[the remark this answers is not in the document]
Comment (elilogbro) [reply spec/jobs/batch_processes/priority_ep_sync_batch_process_job_spec.rb:8]: I left it 99 in order to test the batch limit of 100. If that's not necessary then happy to lower the number!

[the remark this answers is not in the document]
Comment (Aaron-Willis) [reply spec/jobs/batch_processes/priority_ep_sync_batch_process_job_spec.rb:8]: @nkirby-va I'm fine if you want to lower the count, should speed up how fast the RSPEC runs.

File: spec/jobs/batch_processes/priority_ep_sync_batch_process_job_spec.rb [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        expect(BatchProcess.second.records_completed).to eq(BatchProcess::BATCH_LIMIT)
+      end
+
+      it "both batch processes have 0 records_failed" do
+        expect(BatchProcess.first.records_failed).to eq(0)
+        expect(BatchProcess.second.records_failed).to eq(0)
+      end
+    end
+
+    context "when the job duration ends before all PriorityEndProductSyncQueue records can be batched" do
```
[the remark this answers is not in the document]
Comment (Aaron-Willis) [reply spec/jobs/batch_processes/priority_ep_sync_batch_process_job_spec.rb:125]: @nkirby-va Yes, please add a this to the testing steps and run that particular test case against both.

File: app/jobs/populate_end_product_sync_queue_job.rb [hunk 3]
```diff
@@ -3,27 +3,45 @@
...
+          next if priority_epes.empty?
 
-        Rails.logger.info("PopulateEndProductSyncQueueJob EPEs processed: #{batch} - Time: #{Time.zone.now}")
+          priority_epes
+        end
+
+        batch ? insert_into_priority_sync_queue(batch) : stop_job(log_no_records_found: true)
+
+        sleep(SLEEP_DURATION)
+      rescue StandardError => error
```
[the remark this answers is not in the document]
Comment (elilogbro) [reply app/jobs/populate_end_product_sync_queue_job.rb:34]: @Aaron-Willis Hes talking about adding
```Rails.logger.error("Error: #{error.inspect}, Job ID: #{JOB_ATTR&.job_id}, Job Time: #{Time.zone.now}")
        capture_exception(error: error,
                          extra: { job_id: JOB_ATTR&.job_id.to_s,
                                   job_time: Time.zone.now.to_s })```

File: spec/jobs/batch_processes/priority_ep_sync_batch_process_job_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Aaron-Willis) [reply spec/jobs/batch_processes/priority_ep_sync_batch_process_job_spec.rb:126]: No JobDurationError, I meant to please add a test case in the testing steps for job_running_past_expected_end_time?

File: app/jobs/populate_end_product_sync_queue_job.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (Aaron-Willis) [reply app/jobs/populate_end_product_sync_queue_job.rb:34]: Yes, Both should handle errors in the same way.

**Pull request closed without merging**